### PR TITLE
set omp thread limit to 1 and modify batch test

### PR DESCRIPTION
**Pull request by sbarratt**
tests pass

### CODE REVIEW

File: diffcp/cone_program.py [hunk 1]
```diff
@@ -1,4 +1,7 @@
 import warnings
+import os
+
+os.environ["OMP_THREAD_LIMIT"] = "1"
```
Comment (bamos) [inline diffcp/cone_program.py]: Hmm, doing this globally could really mess with other libraries that need to use openmp, what happens if you set it back to whatever it was before after importing SCS here?

Comment (sbarratt) [reply diffcp/cone_program.py]: That unfortunately doesn't work.

Comment (sbarratt) [reply diffcp/cone_program.py]: `mkl.set_num_threads(1)` seems to work. (https://pypi.org/project/mkl/)

**Comment by bamos** [comment]
Hmm, too bad resetting OMP_NUM_THREADS after the SCS import doesn't work. It would be nice to do this without adding a dependency, but this package seems like it does what we want: https://github.com/joblib/threadpoolctl

**Comment by sbarratt** [comment]
> Hmm, too bad resetting OMP_NUM_THREADS after the SCS import doesn't work. It would be nice to do this without adding a dependency, but this package seems like it does what we want: https://github.com/joblib/threadpoolctl

works!
```bash
001 | 3.5063 +/- 0.00 | 0.4420 +/- 0.00
002 | 1.7639 +/- 0.00 | 0.2543 +/- 0.00
003 | 1.3252 +/- 0.00 | 0.1581 +/- 0.00
004 | 0.9329 +/- 0.00 | 0.1308 +/- 0.00
```

**Comment by bamos** [comment]
Awesome!

**Comment by akshayka** [comment]
Sweet!

**Pull request merged**